src/restic_fmt_plug.c

```diff
@@ -0,0 +1,334 @@
+}
+
+int restic_cmp_exact(char *source, int index)
+{
+	return 1;
+}
+
+static int restic_crypt_all(int *pcount, struct db_salt *salt)
+{
+	int count = *pcount;
+	int index;
+	int failed = 0;
+	yescrypt_params_t params = {.N = cur_salt->N, .r = cur_salt->r, .p = cur_salt->p};
+#ifdef _OPENMP
+#pragma omp parallel for
+#endif
+	for (index = 0; index < count; index += MIN_KEYS_PER_CRYPT) {
+		uint8_t kdf_out[64];
+#ifdef _OPENMP
+		int t = omp_get_thread_num();
+		if (t >= max_threads) {
+#pragma omp atomic
+			failed |= 2;
+			continue;
+		}
+#else
+		const int t = 0;
+#endif
+		if (yescrypt_kdf(NULL, &local[t], (const uint8_t *)saved_key[index], strlen(saved_key[index]),
+				 (const uint8_t *)cur_salt->salt, sizeof(cur_salt->salt), &params, kdf_out,
+				 sizeof(kdf_out))) {
+			failed |= 1;
+		}
+		uint8_t *poly1305_key = kdf_out + 32;
+		const unsigned char *nonce = cur_salt->data;
+		const unsigned char *ciphertext = cur_salt->data + NONCE_SIZE;
+		unsigned char prepared_key[32];
+		unsigned char key_mask[16] = {0xff, 0xff, 0xff, 0x0f, 0xfc, 0xff, 0xff, 0x0f,
+					      0xfc, 0xff, 0xff, 0x0f, 0xfc, 0xff, 0xff, 0x0f};
+		unsigned char masked_key[32];
+		int i;
+		memcpy(masked_key, poly1305_key, 32);
+		for (i = 0; i < 16; i++) {
+			masked_key[i + 16] &= key_mask[i];
+		}
+
+		AES_KEY aeskey;
+		AES_set_encrypt_key(masked_key, 128, &aeskey);
+		AES_ecb_encrypt(nonce, prepared_key + 16, &aeskey, AES_ENCRYPT);
+		memcpy(prepared_key, masked_key + 16, 16);
+		poly1305_auth((unsigned char *)crypt_out[index], ciphertext, 128, prepared_key);
+	}
+	if (failed) {
+#ifdef _OPENMP
+		if (failed & 2) {
+			fprintf(stderr, "OpenMP thread number out of range\n");
+			error();
+		}
+#endif
+		fprintf(stderr, "scrypt failed: %s\n", strerror(errno));
+		error();
+	}
+	return count;
+}
+
+static struct fmt_tests restic_tests[] = {
+    {"$restic$scrypt*8192*8*1*"
+     "ed29ad65948797a275f15b1da36fdd2b0247a7772d69ecfdf21141d837fb0780b6fb48cf7ccc3e4146a105dee0df4851256e204671d97c718"
+     "c7e6b4a7a8cfb75*"
+     "879acc157daa013218fcccf6b60be20f1e52baa893698a589f026165f51bbb1da03b0a5db42885d8bcffb34030b0bb26716e9f7c950cccb67"
+     "4494d63d104ee8808e713a7d483a6a0ef36b14aaac652eaa3a92b12f9d7a4cfead72ed0a216ccaeb0ddf9c6e94aa84c82590ae9a6ffc1b48b"
+     "4fba163635ffb4e0633de668827e567e8c834539ae18d750be6f8f86f12101b04ab926fa570038eb6f78ef6021c1b1",
+     "penance"},
+    {"$restic$scrypt*32768*8*5*"
+     "e6dac36997999525fdeeb075434416047dab7d28b0c26f82fa44a469747659ecb86614e66b75151e0bd8acc2f8c7b29d258bf8e6a39d47455"
+     "bd8d34e87622510*"
+     "f82d75db1dc5087fb2bdb014bb51cf07b72b62d10cb164fb244b0358b9c1bbac3663f0ca3f00f12731dc3cc3b6c9de949dc74b31d22e515e5"
+     "5096d219d75babc798a9a7a47fc86034e6fe358a93a6b1eff71d2b791ecee3b775b2b910ecf55abef7d53e4a0fac08e5526f1c25eab06dcd2"
+     "6a1250cb37b7dc115391c680becce8a5e63bdbe2326626c88d16bece0d6dca51de89e43207ea6febc6e119486024a0",
+     "password"},
+    {NULL}};
```

Comment:
We typically keep the test vectors near the start of a format source file, before the first function. And the formatting here is very different from what we use.